pkg/vsphere/tasks/waiter.go

```diff
@@ -89,6 +102,56 @@ func WaitForResult(ctx context.Context, f func(context.Context) (Task, error)) (
 	}
 }
 
+func fixTask(ctx context.Context, sess *session.Session, target object.Reference) error {
+	switch vmm := target.(type) {
+	case *object.VirtualMachine:
+		vm := vm.NewVirtualMachine(ctx, sess, target.Reference())
```

Comment:
I agree with it, but kind of, we're using session widely in this way, for all packages talk to govmomi.
This is a reasonable claim drive us to refactor session I think, but I would prefer to do it separately.